freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/model/UsersStateDifference.java

```diff
@@ -113,6 +125,39 @@ public static UsersStateDifference forDeletedUser(String deletedUser, Collection
         return usersToAdd;
     }
 
+    public static ImmutableSet<String> calculateUsersWithCredentialsToUpdate(UmsUsersState umsState, UsersState ipaState,
+            boolean credentialsUpdateOptimizationEnabled) {
+        ImmutableSet<String> usersWithCredentialsToUpdate;
```

Comment:
after moving to methods it could be simplified like:
```
ImmutableSet<String> usersWithCredentialsToUpdate = credentialsUpdateOptimizationEnabled ? calcUsersWithUpdate(umsState, ipaState) : allUsers(umsState);

Reply:
Done.